### PR TITLE
Enable Part B data

### PR DESCRIPTION
**Pull request by achocron**
Things to consider before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description
Update Part B validation to allow setting baseType/baseData.

### CODE REVIEW

File: AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m [hunk 1]
```diff
@@ -98,11 +98,49 @@ - (void)setPropertiesAndMetadataForCSLog:(MSCommonSchemaLog *)csLog {
   if (self.typedProperties) {
     csProperties = [NSMutableDictionary new];
     metadata = [NSMutableDictionary new];
+    NSString *baseDataPrefix = [NSString stringWithFormat:@"%@.", kMSDataBaseData];
```
Comment (jaelim-ms) [inline AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m]: Synced offline. We should validate baseType and baseData and then remove properties if one of them is missing.

File: AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m [hunk 2]
```diff
@@ -98,11 +98,46 @@ - (void)setPropertiesAndMetadataForCSLog:(MSCommonSchemaLog *)csLog {
...
+
+    // If there is no valid "baseType" property, there must not be "baseData.*" properties.
+    if (![self.typedProperties.properties[kMSDataBaseType] isKindOfClass:[MSStringTypedProperty class]]) {
```
Comment (jaelim-ms) [inline AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m]: nit: can you swap if block and else block by removing `!` from the condition?
```
if (!condition) {
  // Do A
} else {
  // Do B
}
```
can be
```
if (condition) {
  // Do B
} else {
  // Do A
}
```

File: AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m [hunk 3]
```diff
@@ -98,11 +98,46 @@ - (void)setPropertiesAndMetadataForCSLog:(MSCommonSchemaLog *)csLog {
...
+      }
+      if (removedBaseData) {
+        MSLogWarning([MSAnalytics logTag], @"baseData was set but baseType is missing or invalid.");
+      }
+
+      // Base type might be set but invalid, so remove it.
+      [self.typedProperties.properties removeObjectForKey:kMSDataBaseType];
+    }
+
+    // If there are no "baseData.*" properties, there must not be a "baseType" property.
```
Comment (jwallra) [inline AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m]: nit: I prefer not to split if/else clauses this way, I would prefer for the comment to be just inside the else block

Comment (guperrot) [reply AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m]: That's not what we do usually and the comment to apply to the block. It should be same level as the other comment.

**Pull request merged**